WcaOnRails/app/models/user.rb

```diff
@@ -1375,4 +1379,89 @@ def can_access_wfc_senior_matters?
   def can_edit_translators?
     can_edit_any_groups? || software_team?
   end
+
+  private def board_role
+    {
+      group: {
+        id: 'board',
+        name: 'WCA Board of Directors',
+        group_type: UserGroup.group_types[:board],
+        is_hidden: false,
+        is_active: true,
+        metadata: {
+          friendly_id: 'board',
+        },
+      },
+      is_active: true,
+      user: self,
+      metadata: {
+        status: 'member',
+      },
+    }
+  end
+
+  def roles
+    roles = UserRole.where(user_id: self.id).to_a # to_a is to convert the ActiveRecord::Relation to an
+    # array, so that we can append roles which are not yet migrated to the new system. This can be
+    # removed once all roles are migrated to the new system.
+
+    if delegate_status.present?
+      roles << delegate_role
+    end
+
+    roles.concat(team_roles)
+
+    if board_member?
+      roles << board_role
+    end
+
+    Team.all_officers.each do |officer_team|
+      if officer_team == Team.chair
+        status = 'chair'
+      elsif officer_team == Team.executive_director
+        status = 'executive_director'
+      elsif officer_team == Team.secretary
+        status = 'secretary'
+      elsif officer_team == Team.vice_chair
+        status = 'vice_chair'
+      end
+      if team_member?(officer_team)
+        roles << {
+          group: {
+            id: 'officers',
+            name: 'WCA Officers',
+            group_type: UserGroup.group_types[:officers],
+            is_hidden: false,
+            is_active: true,
+          },
+          is_active: true,
+          user: self,
+          metadata: {
+            status: status,
+          },
+        }
+      end
+    end
+
+    if Team.wfc.current_members.select(&:team_leader).map(&:user).include?(self)
```

Comment:
Surprised that a Team doesn't have a `def leader` or `def leaders` method. Can you create one without much effort?

Reply:
Oops there existed, used that now. :-)